pytest_mpi/detect_mpiexec_implementation.py

```diff
@@ -0,0 +1,21 @@
+import subprocess
+
+def detect_mpiexec_implementation():
+    try:
+        result = subprocess.run(
+            ["mpiexec", "--version"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True,
+            check=True
+        )
+        output = result.stdout.lower()
+
+        if "open mpi" in output or "open-rte" in output:
+            return "Open MPI"
+        elif "mpich" in output:
+            return "MPICH"
+        else:
+            return "Unknown MPI implementation"
+    except FileNotFoundError:
+        raise FileNotFoundError("mpiexec not found in PATH")
```

Comment:
```suggestion
        raise FileNotFoundError("mpiexec not found in PATH. Please run in non-forking mode to specify a different MPI executable.")
```
The message should tell the user how to correct the error.

I'm hesitant to just error in this case because it will cause incorrect failures in the case that someone has an MPI distribution without `mpiexec`. I'm thinking specifically of HPC systems where they are likely to have `srun` or `aprun`, or something else depending on the scheduling manager. There's also `mpirun` but I think that's only MPICH and OpenMPI and is deprecated anyway.

The proper fix should should probably go in another PR, but I wanted to just draw attention to it here.
I think the right thing to do here is allow the user to pass the MPI executable, as in the third point in #16, and only error if _both_ no executable is given _and_ we cannot find `mpiexec`.. It's probably unlikely that someone would be running "on the inside" on an HPC because you have way less control, but users are users.

Reply:
I agree that we should allow the user to specify a custom `mpiexec` equivalent command, but I don't think that that needs to be thought about in this PR. Bear in mind that HPC systems would be expected to run with MPI on the outside, at least initially, and so they won't hit this.